packages/flutter/lib/src/material/material_localizations.dart

```diff
@@ -183,7 +183,7 @@ abstract class MaterialLocalizations {
   /// contains a complete set of properties needed to style a [Text] widget.
   ///
   /// See also: https://material.io/go/design-typography
-  TextTheme get localTextGeometry;
+  TextTheme localTextGeometry({ int version = 0 });
```

Comment:
this is a breaking change (since they're expected to implement this, not call it)... but more importantly, i don't think this API change makes much sense. app devs aren't going to care what version it is, they're just going to provide their text theme for their localizations.

but I don't have a good alternative, assuming that we do still want to have different font sizes for each part of the text theme in different locales...

Reply:
The 3 script categories are defined here:
https://material.io/design/typography/language-support.html#language-categories-reference

The adjustments to dense and tall scripts are based on:
> Type sizes smaller than title styles should make adjustments to the Latin type scale. In our dense theme, they're one logical pixel larger.

In our tall and dense themes the smaller than "title" font sizes have been increased by one.

For tall scripts:
> Use regular weight, as medium weight is unavailable in Noto. Avoid using the bold weight, as bold is too heavy.

We replaced the medium weights in the tall theme: FontWeight.w500 => with FontWeight.w400.

The dense text theme uses  TextBaseline.ideographic instead of TextBaseline.alphabetic.